kyuubi-server/src/main/scala/org/apache/kyuubi/session/KyuubiSessionImpl.scala

```diff
@@ -98,6 +98,19 @@ class KyuubiSessionImpl(
 
   private[kyuubi] def openEngineSession(extraEngineLog: Option[OperationLog] = None): Unit = {
     withDiscoveryClient(sessionConf) { discoveryClient =>
+      val engineCredentials = renewEngineCredentials()
+      var openEngineSessionConf = optimizedConf
+      if (engineCredentials.nonEmpty) {
+        sessionConf.set(KYUUBI_ENGINE_CREDENTIALS_KEY, engineCredentials)
+        openEngineSessionConf =
+          optimizedConf ++ Map(KYUUBI_ENGINE_CREDENTIALS_KEY -> engineCredentials)
+      }
+
+      engine = new EngineRef(
+        sessionConf,
```

Comment:
how about this?

@ulysses-you 
we need transfer the config item to EngineRef

Reply:
can we poll out renewEngineCredentials ? so we do not need change the initialization of EngineRef. Then the ordering looks like:

renewEngineCredentials -> new EngineRef -> openEngineSession